src/components/CaptureDetailDialog.js

```diff
@@ -85,7 +88,10 @@ function CaptureDetailDialog(props) {
 
   useEffect(() => {
     cdContext.getCaptureDetail(capture.id);
-
+    const filterSpecies = capture.speciesId ? speciesContext.speciesList.filter(spec => spec.id === capture.speciesId) : [];
```

Comment:
Instead of `speciesList.filter()` here, use `speciesList.find()` with the same comparison function.
It will return the first match, or `null` if there are no matches.